### PR TITLE
Migration testing

### PR DESCRIPTION
**Pull request by oysteoh**
functionality like @dotfloat described further down in a comment

- (ert_storage.db is older than alembic head) Copy db to "backups/{alembic-revision}_ert_storage.db, do the update.
- (ert_storage.db is newer than alembic head)
  - (backups/{alembic-revision}_ert_storage.db is same or older than alembic head) Tell user: "Your version of ERT is too old for this database, but you have a compatible backup here. Copy it if you dare"
  - (backup is also newer than alembic head) Tell user: "Your version of ERT is too old for this database. Please update ERT or delete and start from scratch"

### CODE REVIEW

**Comment by oysteoh** [comment]
test this please!

**Comment by dotfloat** [comment]
So I was thinking the following:

The active database is always `ert_storage.db`

When the current version of `ert_storage.db` and `ert` differ then:
1. (`ert_storage.db` is older than alembic head) Copy db to `"backups/${KOMODO_RELEASE}_ert_storage.db`, do the update.
2. (`ert_storage.db` is newer than alembic head)
    1. (`backups/${KOMODO_RELEASE}_ert_storage.db` is same or older than alembic head) Tell user: `"Your current database is too new, but you have a backup here. Copy it if you dare"`
    2. (backup is also newer than alembic head) Tell user: `"Your version of ERT is too old for this database. Please update ERT or delete and start from scratch"`

File: tests/storage/test_migrations.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        assert os.path.isfile(storage._db_file_name)
+        assert os.path.isfile(
+            "{}/{}_{}".format(
+                storage._backup_folder, initial_revision, storage._db_file_name
+            )
+        )
+
+
+def test_ert_too_old_with_backup(tmpdir):
+
```
Comment (dotfloat) [inline tests/storage/test_migrations.py:55]: remove space

**Review by dotfloat — COMMENTED** [review]
Looks good otherwise

**Review by dotfloat — APPROVED** [review]
:tada:

**Pull request merged**